elephant/test/test_statistics.py

```diff
@@ -842,6 +846,42 @@ def test_annotations(self):
         self.assertIn('kernel', rate.annotations)
         self.assertEqual(rate.annotations['kernel'], kernel_annotation)
 
+    def test_border_correction(self):
+        np.random.seed(0)
+        n_spiketrains = 75
+        rate = 50. * pq.Hz
+        t_start = 0. * pq.ms
+        t_stop = 1000. * pq.ms
+
+        sampling_period = 0.1 * pq.ms
+
+        trial_list = [homogeneous_poisson_process(
+            rate=rate, t_start=t_start,
+            t_stop=t_stop) for _ in range(n_spiketrains)]
+
+        for CORRECTION in (True, False):
+            rates = []
+            for trial in trial_list:
+                # calculate instaneous rate, discard extra dimension
+                instantenous_rate = statistics.instantaneous_rate(
+                    spiketrains=trial,
+                    sampling_period=sampling_period,
+                    kernel='auto',
+                    border_correction=CORRECTION
+                )
+                rates.append(instantenous_rate)
+            rate_estimated = np.mean(rates, axis=0)[:, 0]
```

Comment:
So, here you're looking at average rate in the first bin, averaged across individual realizations, right? I am not sure I fully get the point of the test that follows then. Maybe one could leave some comments to make it easier to follow.

Reply:
I'm just looking over the average rate profile. This indexing is just necessary due to some weird behavior when applying np.mean to a List of Analog-Signal, where a get an array out of shape (N, 1) instead of a true one-dimensional array. But I added some comments to make it easier to understand.